internal/backend/remote-state/pg/backend_test.go

```diff
@@ -307,47 +410,77 @@ func TestBackendConfigSkipOptions(t *testing.T) {
 	for _, tc := range testCases {
 		t.Run(tc.Name, func(t *testing.T) {
 			schemaName := tc.Name
+			tableName := tc.Name
+			indexName := tc.Name
 
 			config := backend.TestWrapConfig(map[string]interface{}{
 				"conn_str":             connStr,
 				"schema_name":          schemaName,
+				"table_name":           tableName,
+				"index_name":           indexName,
 				"skip_schema_creation": tc.SkipSchemaCreation,
 				"skip_table_creation":  tc.SkipTableCreation,
 				"skip_index_creation":  tc.SkipIndexCreation,
 			})
-			schemaName = pq.QuoteIdentifier(schemaName)
+
 			db, err := sql.Open("postgres", connStr)
 			if err != nil {
 				t.Fatal(err)
 			}
 
 			if tc.Setup != nil {
-				tc.Setup(t, db, schemaName)
+				tc.Setup(t, db, schemaName, tableName, indexName)
 			}
-			defer dropSchemaByQuotedName(t, db, schemaName)
+			defer dropSchema(t, db, schemaName)
 
-			b := backend.TestBackendConfig(t, New(encryption.StateEncryptionDisabled()), config).(*Backend)
+			unconfiguredBackend := New(encryption.StateEncryptionDisabled())
+			b := backend.TestBackendConfig(t, unconfiguredBackend, config).(*Backend)
 
 			if b == nil {
 				t.Fatal("Backend could not be configured")
 			}
 
 			// Make sure everything has been created
+			if tc.TestSchemaIsPresent {
+				// Make sure schema exists
+				var count int
+				query := `select count(*) from information_schema.schemata where schema_name=$1`
+				if err = b.db.QueryRow(query, schemaName).Scan(&count); err != nil {
+					t.Fatal(err)
+				}
 
-			// This tests that both the schema and the table have been created
-			_, err = b.db.Query(fmt.Sprintf("SELECT name, data FROM %s.%s LIMIT 1", schemaName, statesTableName))
-			if err != nil {
-				t.Fatal(err)
+				if count != 1 {
+					t.Fatalf("The schema has not been created (%d)", count)
+				}
 			}
+
+			if tc.TestTableIsPresent {
+				// Make sure that the index exists
+				var count int
+
+				query := `select count(*) from pg_catalog.pg_tables where schemaname=$1 and tablename=$2;`
+				err = b.db.QueryRow(query, schemaName, tableName).Scan(&count)
+				if err != nil {
+					t.Fatal(err)
+				}
+
+				if count != 1 {
+					t.Fatalf("The table has not been created (%d)", count)
+				}
+			}
+
 			if tc.TestIndexIsPresent {
 				// Make sure that the index exists
-				query := `select count(*) from pg_indexes where schemaname=$1 and tablename=$2 and indexname=$3;`
 				var count int
-				if err := b.db.QueryRow(query, tc.Name, statesTableName, statesIndexName).Scan(&count); err != nil {
+
+				query := `select count(*) from pg_indexes where schemaname=$1 and tablename=$2 and indexname=$3;`
+				err = b.db.QueryRow(query, schemaName, tableName, indexName+"_name_key").Scan(&count)
+				if err != nil {
 					t.Fatal(err)
 				}
+
 				if count != 1 {
-					t.Fatalf("The index has not been created (%d)", count)
+					t.Fatalf("The index has not been c reated (%d)", count)
```

Comment:
small typo `created`

Reply:
Done! I also fixed a typo in the change log.